### PR TITLE
Fix assorted json errors

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: Bugfixes "Fix assorted json errors"

#### Purpose of change
I have written some code to verify that when parsing json files we actually visit every member of every json object.  Failing to do so is often a sign of a mistake in the json.  For example a typo in a key name, or a misplaced bracket, or a simple misunderstanding of the format.

This happens a lot.  See #33144 #33328 #33502 #33613 #33852 for examples of fixes of this type.  I'd like to catch them earlier.

Enabling that code surfaced numerous errors in the existing json.  I need to fix those errors before I can PR the code that checks for them.

This is the first batch of such fixes, for the dda core game data.

#### Describe the solution
The vast majority of these changes are to the game data itself.  These are:
* Removing parts that just don't do anything and probably aren't important.
* Fixing typos and misplacements when I was able to do so.
* Normalizing the comment format.  Now all json 'comments' must have a key starting with `//`.
* I added a new item group `allclothes_damaged` because it was required for many places which were trying to apply damage in a way that doesn't work.
* Some fields which don't work but contained potentially useful information (like freezing points) I 'commented out' by prefixing them with `//` rather than just deleting them.

I also tweaked a couple of pieces of code:
* Expand vehicle part installation requirements to allow for using both `using` and the 'standard' requirements fields.
* Improved some error messages regarding json malformedness.

#### Describe alternatives you've considered
One of the more concerning changes is that a bunch of magen code is using `chance` in places where that doesn't work.  I considered making it work instead of removing it as I have, but I didn't want to get any further sidetracked in this PR which has already taken a couple of weeks.  All those changers are packaged up in one commit so it can be easily reverted in the future if anyone ever adds support for `chance` in these extra mapgen components.  I probably should have don the commenting out trick.  I can go back and do that if it seems worthwhile.

#### Additional context
If reviewing these changes I recommend doing so one commit at a time, rather than reading throught the diff in aggregate; it will be easier to see what's going on.

Next step is to look for similar errors in (some of) the core mods, and then push the verification code that checks for these mistakes to prevent them returning.

### CODE REVIEW

File: data/json/items/chemicals_and_resources.json [hunk 1]
```diff
@@ -567,10 +567,9 @@
     "weight": "460 g",
     "bashing": 1,
     "ammo_type": "components",
-    "charges": 2,
     "phase": "liquid",
     "container": "bottle_glass",
-    "freezing_point": 25,
+    "//freezing_point": 25,
```
Comment (ZhilkinSerg) [inline data/json/items/chemicals_and_resources.json]: What's the idea here? Isn't `freezing_point` a valid node?

File: data/json/items/chemicals_and_resources.json [hunk 2]
```diff
@@ -958,7 +955,7 @@
     "name": "handful of hickory nuts",
     "name_plural": "handfuls of hickory nuts",
     "description": "A handful of hard nuts from a hickory tree, still in their shell.",
-    "spoils_in": "180 days",
+    "//spoils_in": "180 days",
```
Comment (ZhilkinSerg) [inline data/json/items/chemicals_and_resources.json]: `spoils_in` should be valid node too.

**Comment by snipercup** [comment]
I tested your changes and encountered no problems. 

- I could build it with msys2 on windows and there were no unusual errors.

In-game:
- Browsed trough recipes
- Browsed trough debug item spawn list
- visited different locations on the map (cities and specials and fields)
- Uninstalled and installed a vehicle part

The only errors I got were:
```
 DEBUG    : invalid furniture id "f_feather_mattress"
 
 FUNCTION : const T& generic_factory<T>::obj(const string_id<T>&) const [with T = furn_t]
 FILE     : src/generic_factory.h
 LINE     : 356



 DEBUG    : Tried to access invalid map position at grid (0,11,0)
 
 FUNCTION : size_t map::get_nonant(const tripoint&) const
 FILE     : src/map.cpp
 LINE     : 8148
```

But I don't think they are related to your change.

**Comment by Night-Pryanik** [comment]
Damn, so many errors. So many things worked not the way they should.

File: data/json/items/chemicals_and_resources.json (hunk 1 above)
Comment (jbytheway) [reply data/json/items/chemicals_and_resources.json]: Freezing point is only valid for comestibles.  This is an ammo.

File: data/json/items/chemicals_and_resources.json (hunk 2 above)
Comment (jbytheway) [reply data/json/items/chemicals_and_resources.json]: `spoils_in` is only valid for comestibles.  This is a generic item.

**Comment by jbytheway** [comment]
The two cases @ZhilkinSerg highlighted are both examples of item fields which are only valid for some types of item.  By default, each item only loads basic info and the details for its slot (in this case, the comestible slot).  For some slots it is possible to use them for other item types by explicitly including a sub-member for that slot (e.g. `"container_data"` for the `islot_container`).  However, that doesn't appear to be possible for `islot_comestible`.

**Pull request merged**